### PR TITLE
objecter: add op latency stat and in-flight count

### PR DESCRIPTION
**Pull request by yuvalif**
this is done in order to better estimate the origin of the end-to-end
operation latency. and see how much of it is due to the objecter and how
much of it is due to the client that uses it (e.g. the RGW)

Signed-off-by: yuval Lifshitz <ylifshit@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by adamemerson — APPROVED** [review]

**Comment by yuvalif** [comment]
currently teuthology is failing on valgrind issues that seems unrelated to the change here:
http://pulpito.front.sepia.ceph.com/yuvalif-2022-06-07_06:57:55-rgw:verify-wip-yuval-objecter-latency-distro-default-smithi/

@adamemerson any idea how can that be related?

File: src/osdc/Objecter.cc [hunk 1]
```diff
@@ -3569,6 +3573,8 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
   ldout(cct, 15) << "handle_osd_op_reply completed tid " << tid << dendl;
   _finish_op(op, 0);
 
+  logger->tinc(l_osdc_op_latency, ceph::coarse_mono_time::clock::now() - op->stamp);
```
Comment (cbodley) [inline src/osdc/Objecter.cc:3576]: regarding the valgrind issues, it may be that `_finish_op(op, 0);` is freeing this `op`?

Comment (yuvalif) [reply src/osdc/Objecter.cc:3576]: you are right. see line 3185. the memory is released. thanks!

**Comment by yuvalif** [comment]
passing teuthology: http://pulpito.front.sepia.ceph.com/yuvalif-2022-06-07_18:27:51-rgw:verify-wip-yuval-objecter-latency-distro-default-smithi/

**Pull request merged**